agriculture/static/js/history.js

```diff
@@ -268,64 +279,122 @@ function drawValveChart(macAddr, refresh=false, showProgress=false) {
             valveData[macAddr] = response.data;
             drawValveChart(macAddr);
             $("#valve-" + macAddr + "-chart-loading").hide();
-        }).fail(function(response) {
-            processErrorResponse(response);
         });
     } else {
         drawChart("valve-" + macAddr + "-chart", valveData[macAddr], "Valve", "Closed/Open", "#0000CC");
     }
 }
 
+//google.charts.load('current', {
+//     callback: drawChart,
+//     packages: ['scatterChart', 'lineChart']
+//});
+
+
 // Draws the chart with the given data.
-function drawChart(id, data, title, units, color=null) {
-    if (!isHistoryShowing())
+function drawChart(id, data, title, units, color=null, data2=null, units2=null, title2=null, color2=null) {
+    if (!isHistoryShowing() || id === undefined)
         return;
 
     var dataTable = new google.visualization.DataTable();
     dataTable.addColumn("date", "");
-    dataTable.addColumn("number", "");
+    dataTable.addColumn("number", title);
+    if (data2 != null){
+        dataTable.addColumn("number", title2);
+    }
 
     if (data.length == 0) {
         $("#" + id).empty();
         $("#" + id).append("<span class='no-data-label'>Not enough data</span>");
         return;
     }
 
-    dataTable.addRows(data.length);
+    var maximumLength = data.length;
+    if (data2 != null && data2.length > maximumLength) {
+        maximumLength = data2.length;
+    }
+
+    dataTable.addRows(maximumLength);
 
     $.each(data, function(k, v) {
         dataTable.setCell(k, 0, new Date(v["timestamp"]));
         dataTable.setCell(k, 1, v["data"]);
     });
 
-    var options = {
-        backgroundColor: "transparent",
-        series: {
-            0: {
-                axis: "Data",
-                color: color,
-                visibleInLegend: false
-            }
-        },
-        axes: {
-            y: {
-                Data: {
-                    label: units
+    if(data2 != null){
+        $.each(data2, function(k, v) {
+            dataTable.setCell(k, 2, v["data"]);
+        });
+    }
+
+    var options = null;
+
+    if(data2 == null && id == "rain-chart"){
+        options = {
+            backgroundColor: "transparent",
+            series: {
+              0: {targetAxisIndex: 0, color: color},
+            },
+            vAxes: {
+              // Adds titles to each axis.
+              0: {title: units},
+            },
+            vAxis: {
+//                viewWindow: {
```

Comment:
It is better to remove comments that are not useful. This only adds complexity to the code.